### PR TITLE
Follow-Up: Unsustainable Biomass

### PR DESCRIPTION
**Pull request by cpschau**
1. Using the unsustainable biomass potentials introduced in #1139, the model becomes infeasible for setups with spatially resolved biomass demand because no statistics are provided for Switzerland in the Eurostat data. Consequently, the country's unsustainable biomass potential cannot be determined and is set to zero. As a result, for networks with deactivated solid biomass transport and zero sustainable biomass potentials, there is no source to cover the unavoidable solid biomass load ("solid biomass for industry"). With this PR, the sustainable biomass potentials for Switzerland are retained in build_biomass_potentials, independent of the configuration setting biomass:share_sustainable_potential_available.
2. The global biomass constraint, which sets the biomass potentials as the upper bound for solid biomass usage in the model within prepare_sector_network, has been adjusted to include the unsustainable potentials.
3. In prepare_sector_network, oil buses are added when incorporating unsustainable bioliquids. Otherwise, there is a risk that the bioliquid-to-oil link may use undefined buses.
4. For overnight scenarios, the biomass potentials are adjusted to include only sustainable sources in build_biomass_potentials.

### CODE REVIEW

**Comment by lisazeyen** [comment]
Great @cpschau!

I think the unsustainable biomass potential should not depend on the foresight and only on your planning horizon. So also for an overnight scenario, it would be good, to take the unsustainable biomass potential defined in the config.

Can you add a plot of the unsustainable potential of each country? Are all other countries, like e.g. Norway and Great Britain covered?

You have a separate carrier unsustainable biomass so you also need a separate Global Constraint for the operational limit I suppose [here](https://github.com/PyPSA/pypsa-eur/blob/06dc60db11357e81f548fa7e813d288697682789/scripts/prepare_sector_network.py#L2630-L2637)

**Comment by cpschau** [comment]
> Great @cpschau!
> 
> I think the unsustainable biomass potential should not depend on the foresight and only on your planning horizon. So also for an overnight scenario, it would be good, to take the unsustainable biomass potential defined in the config.
> 
> Can you add a plot of the unsustainable potential of each country? Are all other countries, like e.g. Norway and Great Britain covered?
> 
> You have a separate carrier unsustainable biomass so you also need a separate Global Constraint for the operational limit I suppose [here](https://github.com/PyPSA/pypsa-eur/blob/06dc60db11357e81f548fa7e813d288697682789/scripts/prepare_sector_network.py#L2630-L2637)

Thanks for the quick review @lisazeyen! Here are the potentials for 2020 (only unsustainable except for Switzerland) and 2050 (only sustainable). Norway and UK are covered in the data.

2020:
biomass_potentials_unsus
2050:
biomass_potentials_sus

I will remove the distinction of the overnight potentials, I do not remember what I was thinking when I added it tbh :D.

Additionally, I'll introduce a global constraint for the operational limit on unsustainable biomass. I believe it's also necessary to add an extra generator for unsustainable biomass in case we decide to spatially resolve the biomass demand without allowing for biomass transport.

File: scripts/prepare_sector_network.py [hunk 1]
```diff
@@ -2631,7 +2648,7 @@ def add_biomass(n, costs):
             "GlobalConstraint",
             "biomass limit",
             carrier_attribute="solid biomass",
-            sense="<=",
+            sense="==",
```
Comment (lindnemi) [inline scripts/prepare_sector_network.py:2651]: this stays

**Comment by lindnemi** [comment]
bitte noch `unsustainable solid biomass: '#998622'` zu 'tech_colors:' hinzufügen

**Review by fneum — APPROVED** [review]

**Pull request merged**